### PR TITLE
feat(socket:ip): Introduce 'socket:ip' module

### PR DESCRIPTION
**Pull request by heapwolf**
This PR introduces the `socket:ip` module and removes the useless `socket:net` module

### CODE REVIEW

**Review by chicoxyzzy — APPROVED** [review]
this needs `npm run gen` to regenerate types

File: api/ip.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * @module IP
+ * Functions for working with IP addresses
+ *
+ */
+
+const v4Seg = '(?:[0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])'
+const v4Str = `(${v4Seg}[.]){3}${v4Seg}`
+const IPv4Reg = new RegExp(`^${v4Str}$`)
+
+export const isIPv4 = s => {
```
Comment (jwerle) [inline api/ip.js:11]: can we rewrite this as:

```js
/**
 * Determines if an input `string` is in IP address version 4 format.
 * @param {string} string
 * @return {boolean}
 * /
export function isIPv4 (string) {
  return typeof string === 'string' && IPv4Reg.test(string)
}
```

File: api/ip.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @module IP
+ * Functions for working with IP addresses
+ *
```
Comment (jwerle) [inline api/ip.js:4]: new line

**Review by jwerle — CHANGES_REQUESTED** [review]
Left some comments. We need to generate docs for this change: `npm run gen`

File: api/ip.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
```
Comment (jwerle) [inline api/ip.js:1]: This module needs a `default` exports as all `socket:` modules require it:

```js
export default { isIPv4 }
```

should work

**Review by jwerle — APPROVED** [review]

**Pull request merged**